Refactor to accommodate different SLSv2 product types.

darora opened the pull request:
- Moves packages; com.palantir.gradle.javadist -> com.palantir.gradle.dist.service
- Moves properties common to all SLSv2 distributions to a `BaseDistributionExtension`
- Functionality change: you can now set `productType` on the extension (vs a hard-coded `service.v1` in the `CreateManifestTask`

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/palantir/gradle-java-distribution/158)
<!-- Reviewable:end -->

Commit:
Refactor to accommodate different SLSv2 product types.

Commit:
Remove unnecessary things

uschi2000 commented:
Review status: 0 of 18 files reviewed at latest revision, 8 unresolved discussions.

---

*[src/main/groovy/com/palantir/gradle/dist/BaseDistributionExtension.groovy, line 7 at r1](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/158#-Kb0RQfl824X1X4mq17D:-Kb0RQfl824X1X4mq17E:b-qen1ug) ([raw file](https://github.com/palantir/gradle-java-distribution/blob/4a894f2663400120309ee935f9fc7b8c2ec62dd4/src/main/groovy/com/palantir/gradle/dist/BaseDistributionExtension.groovy#L7)):*
> ```Groovy
> class BaseDistributionExtension {
> 
>     private static final List<String> VALID_PRODUCT_TYPES = [
> ```

nit: Set

---

*[src/main/groovy/com/palantir/gradle/dist/BaseDistributionExtension.groovy, line 9 at r1](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/158#-Kb0RupNy4RkhBntumpf:-Kb0RupNy4RkhBntumpg:b-5scpu2) ([raw file](https://github.com/palantir/gradle-java-distribution/blob/4a894f2663400120309ee935f9fc7b8c2ec62dd4/src/main/groovy/com/palantir/gradle/dist/BaseDistributionExtension.groovy#L9)):*
> ```Groovy
>     private static final List<String> VALID_PRODUCT_TYPES = [
>             "service.v1",
>             "asset.v1",
> ```

add asset and daemon in subsequent PR, please. this one should be just a refactoring.

---

*[src/main/groovy/com/palantir/gradle/dist/BaseDistributionExtension.groovy, line 20 at r1](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/158#-Kb0RHX7YOEF9q49tcDT:-Kb0RHX7YOEF9q49tcDU:b8mfdr0) ([raw file](https://github.com/palantir/gradle-java-distribution/blob/4a894f2663400120309ee935f9fc7b8c2ec62dd4/src/main/groovy/com/palantir/gradle/dist/BaseDistributionExtension.groovy#L20)):*
> ```Groovy
> 
>     public BaseDistributionExtension(Project project) {
>         this.project = project;
> ```

Either make this a Java file or remove all `;`

---

*[src/main/groovy/com/palantir/gradle/dist/BaseDistributionExtension.groovy, line 23 at r1](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/158#-Kb0RXetFJ472G_4tXHs:-Kb0RXetFJ472G_4tXHt:bdy0ylx) ([raw file](https://github.com/palantir/gradle-java-distribution/blob/4a894f2663400120309ee935f9fc7b8c2ec62dd4/src/main/groovy/com/palantir/gradle/dist/BaseDistributionExtension.groovy#L23)):*
> ```Groovy
>     }
> 
>     public void serviceName(String serviceName) {
> ```

prefer logical grouping of setters/getters: either group all setters and then all getters, or group corresponding setter/getter

---

*[src/main/groovy/com/palantir/gradle/dist/BaseDistributionExtension.groovy, line 51 at r1](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/158#-Kb0RkNDtlbZMHVwf5Rw:-Kb0RkNEeP-Ifdy-L9QG:b-t8ou5f) ([raw file](https://github.com/palantir/gradle-java-distribution/blob/4a894f2663400120309ee935f9fc7b8c2ec62dd4/src/main/groovy/com/palantir/gradle/dist/BaseDistributionExtension.groovy#L51)):*
> ```Groovy
>     }
> 
>     public String getProductType() {
> ```

is this a `distributionType` or a `packageType`? think I prefer `distributionType`?
ah, it's because it's "product-type" in the sls spec and in the manifest? cool, then we should leave it as product-type.

---

*[src/main/groovy/com/palantir/gradle/dist/BaseDistributionExtension.groovy, line 57 at r1](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/158#-Kb0RT0rS68aRSOEeM6r:-Kb0RT0rS68aRSOEeM6s:b8femm4) ([raw file](https://github.com/palantir/gradle-java-distribution/blob/4a894f2663400120309ee935f9fc7b8c2ec62dd4/src/main/groovy/com/palantir/gradle/dist/BaseDistributionExtension.groovy#L57)):*
> ```Groovy
>     public void setProductType(String type) {
>         if (!VALID_PRODUCT_TYPES.contains(type)) {
>             throw new IllegalArgumentException("Invalid product type specified: " + type)
> ```

print know types

---

*[src/main/groovy/com/palantir/gradle/dist/service/ServiceDistributionExtension.groovy, line 21 at r1](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/158#-Kb0SJMVwLzQud33sTDc:-Kb0SJMVwLzQud33sTDd:bxiwv08) ([raw file](https://github.com/palantir/gradle-java-distribution/blob/4a894f2663400120309ee935f9fc7b8c2ec62dd4/src/main/groovy/com/palantir/gradle/dist/service/ServiceDistributionExtension.groovy#L21)):*
> ```Groovy
> import org.gradle.api.Project;
> 
> class ServiceDistributionExtension extends BaseDistributionExtension {
> ```

it's not quite clear to me why you need the BaseExt and the ServiceExt, but I guess this will become clear in a follow-up?

---

*[src/test/groovy/com/palantir/gradle/dist/BaseDistributionExtensionTest.groovy, line 29 at r1](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/158#-Kb0TNUUua3CZDHmLbbx:-Kb0TNUUua3CZDHmLbby:b-zemuq0) ([raw file](https://github.com/palantir/gradle-java-distribution/blob/4a894f2663400120309ee935f9fc7b8c2ec62dd4/src/test/groovy/com/palantir/gradle/dist/BaseDistributionExtensionTest.groovy#L29)):*
> ```Groovy
>     }
> 
>     def 'productType uses service as default'() {
> ```

that seems wrong: the BaseExt shouldn't have a bias, IMO.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/158)*
<!-- Sent from Reviewable.io -->

uschi2000 commented:
Reviewed 33 of 33 files at r1.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/158)*
<!-- Sent from Reviewable.io -->

Commit:
Address CR

darora commented:
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[src/main/groovy/com/palantir/gradle/dist/BaseDistributionExtension.groovy, line 7 at r1](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/158#-Kb0RQfl824X1X4mq17D:-Kb0WcqEj9nc0LgJo1mH:bb7f2mf) ([raw file](https://github.com/palantir/gradle-java-distribution/blob/4a894f2663400120309ee935f9fc7b8c2ec62dd4/src/main/groovy/com/palantir/gradle/dist/BaseDistributionExtension.groovy#L7)):*
<details><summary><i>Previously, uschi2000 (Robert Fink) wrote…</i></summary><blockquote>

nit: Set
</blockquote></details>

done

---

*[src/main/groovy/com/palantir/gradle/dist/BaseDistributionExtension.groovy, line 9 at r1](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/158#-Kb0RupNy4RkhBntumpf:-Kb0WbNS2bUOeLludqjw:b-hfpv0l) ([raw file](https://github.com/palantir/gradle-java-distribution/blob/4a894f2663400120309ee935f9fc7b8c2ec62dd4/src/main/groovy/com/palantir/gradle/dist/BaseDistributionExtension.groovy#L9)):*
<details><summary><i>Previously, uschi2000 (Robert Fink) wrote…</i></summary><blockquote>

add asset and daemon in subsequent PR, please. this one should be just a refactoring.
</blockquote></details>

removed

---

*[src/main/groovy/com/palantir/gradle/dist/BaseDistributionExtension.groovy, line 20 at r1](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/158#-Kb0RHX7YOEF9q49tcDT:-Kb0WZjGbPyxRUqxNJUh:bb7f2mf) ([raw file](https://github.com/palantir/gradle-java-distribution/blob/4a894f2663400120309ee935f9fc7b8c2ec62dd4/src/main/groovy/com/palantir/gradle/dist/BaseDistributionExtension.groovy#L20)):*
<details><summary><i>Previously, uschi2000 (Robert Fink) wrote…</i></summary><blockquote>

Either make this a Java file or remove all `;`
</blockquote></details>

done

---

*[src/main/groovy/com/palantir/gradle/dist/BaseDistributionExtension.groovy, line 23 at r1](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/158#-Kb0RXetFJ472G_4tXHs:-Kb0WabYy8aRM-yicXUO:bb7f2mf) ([raw file](https://github.com/palantir/gradle-java-distribution/blob/4a894f2663400120309ee935f9fc7b8c2ec62dd4/src/main/groovy/com/palantir/gradle/dist/BaseDistributionExtension.groovy#L23)):*
<details><summary><i>Previously, uschi2000 (Robert Fink) wrote…</i></summary><blockquote>

prefer logical grouping of setters/getters: either group all setters and then all getters, or group corresponding setter/getter
</blockquote></details>

done

---

*[src/main/groovy/com/palantir/gradle/dist/BaseDistributionExtension.groovy, line 51 at r1](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/158#-Kb0RkNDtlbZMHVwf5Rw:-Kb0WdyXUQxekqGhytAg:bb7slsi) ([raw file](https://github.com/palantir/gradle-java-distribution/blob/4a894f2663400120309ee935f9fc7b8c2ec62dd4/src/main/groovy/com/palantir/gradle/dist/BaseDistributionExtension.groovy#L51)):*
<details><summary><i>Previously, uschi2000 (Robert Fink) wrote…</i></summary><blockquote>

is this a `distributionType` or a `packageType`? think I prefer `distributionType`?
ah, it's because it's "product-type" in the sls spec and in the manifest? cool, then we should leave it as product-type.
</blockquote></details>

yup!

---

*[src/main/groovy/com/palantir/gradle/dist/BaseDistributionExtension.groovy, line 57 at r1](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/158#-Kb0RT0rS68aRSOEeM6r:-Kb0WeqFEWGsNmCJGiB8:bb7f2mf) ([raw file](https://github.com/palantir/gradle-java-distribution/blob/4a894f2663400120309ee935f9fc7b8c2ec62dd4/src/main/groovy/com/palantir/gradle/dist/BaseDistributionExtension.groovy#L57)):*
<details><summary><i>Previously, uschi2000 (Robert Fink) wrote…</i></summary><blockquote>

print know types
</blockquote></details>

done

---

*[src/main/groovy/com/palantir/gradle/dist/service/ServiceDistributionExtension.groovy, line 21 at r1](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/158#-Kb0SJMVwLzQud33sTDc:-Kb0Wfb7Au02yqcoOsW2:b-97ryph) ([raw file](https://github.com/palantir/gradle-java-distribution/blob/4a894f2663400120309ee935f9fc7b8c2ec62dd4/src/main/groovy/com/palantir/gradle/dist/service/ServiceDistributionExtension.groovy#L21)):*
<details><summary><i>Previously, uschi2000 (Robert Fink) wrote…</i></summary><blockquote>

it's not quite clear to me why you need the BaseExt and the ServiceExt, but I guess this will become clear in a follow-up?
</blockquote></details>

yup! I'll be adding a separate extension for assets which, aside from the shared properties in BaseExt, will present a disjoint set of functionality vis-a-vis the considerable configurability of ServiceExt.

Not opposed to keeping a single extension and just adding an `assetDirs` or similar, but imo would be fairly confusing to have a large number of knobs that are ignored based on the plugin in use

---

*[src/test/groovy/com/palantir/gradle/dist/BaseDistributionExtensionTest.groovy, line 29 at r1](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/158#-Kb0TNUUua3CZDHmLbbx:-Kb0Wke3YXU1jX0qVk-W:b-oasuuc) ([raw file](https://github.com/palantir/gradle-java-distribution/blob/4a894f2663400120309ee935f9fc7b8c2ec62dd4/src/test/groovy/com/palantir/gradle/dist/BaseDistributionExtensionTest.groovy#L29)):*
<details><summary><i>Previously, uschi2000 (Robert Fink) wrote…</i></summary><blockquote>

that seems wrong: the BaseExt shouldn't have a bias, IMO.
</blockquote></details>

removed default

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/158)*
<!-- Sent from Reviewable.io -->

Commit:
More semicolons :|

uschi2000 commented:
Reviewed 1 of 33 files at r1, 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/main/groovy/com/palantir/gradle/dist/service/ServiceDistributionExtension.groovy, line 21 at r1](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/158#-Kb0SJMVwLzQud33sTDc:-Kb9pVtpkCu_CInrJ3fc:bb7efny) ([raw file](https://github.com/palantir/gradle-java-distribution/blob/4a894f2663400120309ee935f9fc7b8c2ec62dd4/src/main/groovy/com/palantir/gradle/dist/service/ServiceDistributionExtension.groovy#L21)):*
<details><summary><i>Previously, darora (Divyanshu Arora) wrote…</i></summary><blockquote>

yup! I'll be adding a separate extension for assets which, aside from the shared properties in BaseExt, will present a disjoint set of functionality vis-a-vis the considerable configurability of ServiceExt.

Not opposed to keeping a single extension and just adding an `assetDirs` or similar, but imo would be fairly confusing to have a large number of knobs that are ignored based on the plugin in use
</blockquote></details>

cool

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/158)*
<!-- Sent from Reviewable.io -->

uschi2000 commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/gradle-java-distribution/158#-:-Kb9ptycMp4eupVurBAr:bnfp4nl)*
<!-- Sent from Reviewable.io -->

uschi2000 commented:
Cheers, Div.
